openlibrary/plugins/openlibrary/js/carousels.js

```diff
@@ -82,40 +80,77 @@ const Carousel = {
                 '"aria-hidden="false" role="option">' +
                 '<div class="book-cover">' +
                   '<a href="' + work.key + '" ' + isClickable + '>' +
-                    '<img class="bookcover" width="130" height="200" title="' + work.title + '" ' +
-                         'src="//covers.openlibrary.org/b/id/' + work.cover_id + '-M.jpg">' +
+                    '<img class="bookcover" width="130" height="200" title="' +
+                      work.title + '" ' +
+                      'src="//covers.openlibrary.org/b/id/' + cover_id + '-M.jpg">' +
                   '</a>' +
                 '</div>' +
                 '<div class="book-cta">' +
-                  '<a class="btn cta-btn ' + cls + '" href="' + url + '" data-ol-link-track="subjects" ' +
-                  'title="' + cta + ': ' + work.title + '" data-key="subjects" data-ocaid="' + ocaid + '">' + cta + '</a>' +
+                  '<a class="btn cta-btn ' + cls + '" href="' + url +
+                    '" data-ol-link-track="subjects" ' +
+                    'title="' + cta + ': ' + work.title +
+                    '" data-key="subjects" data-ocaid="' + ocaid + '">' + cta +
+                  '</a>' +
                 '</div>' +
               '</div>';
         }
 
-        $('.carousel-'+key).on('afterChange', function() {
-            var totalSlides = $('.carousel-' + key + '.slick-slider').slick("getSlick").$slides.length;
-            var numActiveSlides = $('.carousel-' + key + ' .slick-active').length;
-            var currentLastSlide = $('.carousel-' + key + '.slick-slider').slick('slickCurrentSlide') + numActiveSlides;
-            // this allows us to pre-load before hitting last page
-            var lastSlideOnSecondToLastPage = (totalSlides - numActiveSlides);
+        // if a loadMore config is provided and it has a (required) url
+        if (loadMore && loadMore.url) {
+            var url;
+            try {
+                // exception handling needed in case loadMore.url is relative path
+                url = new URL(loadMore.url);
+            } catch (e) {
+                url = new URL(window.location.origin + loadMore.url);
+            }
+            var default_limit = 18; // 3 pages of 6 books
+            url.searchParams.set('limit', loadMore.limit || default_limit);
+            loadMore.pageMode = loadMore.pageMode === 'page' ? 'page' : 'offset'; // verify pagination mode
+            loadMore.locked = false; // prevent additional calls when not in critical section
+
+            // Bind an action listener to this carousel on resize or advance
+            $(selector).on('afterChange', function() {
```

Comment:
I'm guessing the afterChange event is added by slick ? If so can you document this with a detailed comment so it's obvious, as it wasn't to me.